src/python/pants/init/extension_loader.py

```diff
@@ -130,6 +130,37 @@ def load_backends_and_plugins(
             deprecation_start_version="1.28.0.dev0",
             hint="\n".join(msg_lines),
         )
+
+    if "pants.contrib.awslambda.python" in backends1:
+        msg_lines = [
+            fill(
+                (
+                    "The original awslambda implementation is being replaced by an improved "
+                    "implementation made possible by the V2 engine."
+                ),
+                80,
+            ),
+            "",
+            fill(
+                (
+                    "To prepare for this change, remove 'pants.contrib.awslambda.python' from the "
+                    "`backend_packages` entry in the `GLOBAL` section in your `pants.toml` "
+                    "(or `pants.ini`). Then add `pants.backend.awslambda.python` to the "
+                    "`backend_packages2` entry. Ensure that you have `--v2` enabled (the default). "
+                    "Then modify your BUILD files to remove the intermediate `python_binary` "
+                    "target and the `binary=` setting on the `python_awslambda` target. "
+                    "Finally, use the `awslambda` goal instead of the `bundle` goal."
```

Comment:
It'd be good to point to the new example repo showing how to do this all. I think this example doesn't yet exist, so totally fine to forward fix.

The reason I emphasize these deprecation messages is that they're one of our best opportunities to get pre-existing users to try out V2. For example, if we can get them to check out those new docs, then they might check out other things and be interested in trying more.

Reply:
Yeah, that example has to wait until I fix up a couple of things in this repo...